Java/core/src/main/java/com/amazon/randomcutforest/state/RandomCutForestState.java

```diff
@@ -48,8 +48,22 @@
 
     private boolean compactEnabled;
 
+    private boolean internalShinglingEnabled;
+
+    private boolean internalRotationEnabled;
+
+    private boolean dynamicResizingEnabled;
```

Comment:
These are really properties of the point store state, right? Are we going to validate that the point store state contained in this forest state object has these properties? If not, then what purpose to these fields have?

Reply:
Fair point. The latter two are indeed details of pointStore and will move them there. Internal shingling however requires visibility from the forest level class -- we should decide whether to augment/or nor to, without a roundtrip. Moreover this flag would enable checks at the input level of the forest.